sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -329,6 +330,9 @@ case class GpuOrcMultiFilePartitionReaderFactory(
             OrcSchemaWrapper(orcPartitionReaderContext.updatedReadSchema),
             OrcExtraInfo(orcPartitionReaderContext.requestedMapping)))
     }
+    if (metrics.contains("scanTime")) {
+      metrics("scanTime") += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - currentTime)
+    }
```

Comment:
This does an unnecessary double-lookup in the map, once for the contains and once again to do the update.
```suggestion
    metrics.get("scanTime").foreach {
      _ += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - currentTime)
    }
```
Applies to the other places contains is used.

Reply:
Done, thanks!